packages/core/src/sync/fragments.ts

```diff
@@ -166,93 +184,71 @@ function parseTopics(topics: (Hex | Hex[] | null)[] | undefined) {
-    toAddress: Address | null;
-  }) => {
-    return `${chainId}_${fromAddress}_${toAddress}`;
-  };
-
-  const factoryIdCallback = ({
-    chainId,
-    fromAddress,
-    address,
-    eventSelector,
-    childAddressLocation,
-  }: Omit<TraceFilterFragment, "id" | "toAddress"> & {
-    address: Address;
-    eventSelector: LogFactory["eventSelector"];
-    childAddressLocation: LogFactory["childAddressLocation"];
-  }) => {
-    return `${chainId}_${address}_${eventSelector}_${childAddressLocation}_${fromAddress}`;
-  };
+}): FragmentReturnType => {
+  const fragments: FragmentReturnType = [];
 
   if (isAddressFactory(toAddress)) {
-    for (const _fromAddress of Array.isArray(fromAddress)
-      ? fromAddress
-      : [null]) {
-      for (const _factoryAddress of Array.isArray(toAddress.address)
+    for (const _fromAddress of fromAddress === undefined
+      ? [null]
+      : fromAddress) {
+      for (const _toAddress of Array.isArray(toAddress.address)
         ? toAddress.address
         : [toAddress.address]) {
-        fragments.push({
-          id: factoryIdCallback({
-            chainId,
-            fromAddress: _fromAddress,
-            address: _factoryAddress,
-            eventSelector: toAddress.eventSelector,
-            childAddressLocation: toAddress.childAddressLocation,
-          }),
-          chainId,
-          address: _factoryAddress,
-          eventSelector: toAddress.eventSelector,
-          childAddressLocation: toAddress.childAddressLocation,
-          fromAddress: _fromAddress,
-        });
+        const id = `${chainId}_${_toAddress}_${toAddress.eventSelector}_${toAddress.childAddressLocation}_${_fromAddress}`;
+
+        const adjacent = [id];
+
+        if (_fromAddress !== null) {
+          adjacent.push(
+            `${chainId}_${_toAddress}_${toAddress.eventSelector}_${toAddress.childAddressLocation}_${null}`,
+          );
+        }
+
+        fragments.push({ id, adjacent });
       }
     }
   } else {
-    for (const _fromAddress of Array.isArray(fromAddress)
-      ? fromAddress
-      : [null]) {
-      for (const _toAddress of Array.isArray(toAddress) ? toAddress : [null]) {
-        fragments.push({
-          id: idCallback({
-            chainId,
-            fromAddress: _fromAddress,
-            toAddress: _toAddress,
-          }),
-          chainId,
-          toAddress: _toAddress,
-          fromAddress: _fromAddress,
-        });
+    for (const _fromAddress of fromAddress === undefined
+      ? [null]
+      : fromAddress) {
+      for (const _toAddress of toAddress === undefined
+        ? [null]
+        : (toAddress as Address[])) {
+        const id = `${chainId}_${_fromAddress}_${_toAddress}`;
+
+        const adjacent = [id];
+
+        if (_toAddress !== null) {
+          adjacent.push(`${chainId}_${null}_${_fromAddress}`);
+        }
+
+        if (_fromAddress !== null) {
+          adjacent.push(`${chainId}_${_toAddress}_${null}`);
+        }
```

Comment:
Is this a typo? Do we expect all the ids to be `${chainId}_${_fromAddress}_${_toAddress}` order for traces?

Reply:
Might be misunderstanding your question. Are you wondering about the lack of factory contracts or the lack of fields like "error" or "function selector"?

Reply:
Or the `null`

Reply:
no, in the first id you have fromAddress_toAddress order, in the second id you have fromAddress second , and in the third id you have toAddress first. Is there a reason behind switched order of addresses with respect to first one?

Reply:
Oh that seems like a typo. Really nice catch - would have taken a while to debug this